tests/execute.spec.js

```diff
@@ -59,7 +38,10 @@ describe('Scripts', function() {
       });
 
       it('should yield the error', function(done) {
-        scripts.run('foo', (error, output) => {
+        execute.extractAndRun({
+          content: '#!/bin/bash\necho "Hello"',
```

Comment:
Given the way that things are being stubbed, I think you should make it clear that the `content` here isn't actually being used, e.g. by changing it to `content: 'dummy content'`.

And given this new extractAndRun mechanism, I think you should add some platform-specific tests that *do* actually test it. (e.g. `if (os.platform === 'win32') { extractAndRun('@echo off\n@echo Hello world\n') } else { '#!/bin/bash\necho "Hello World"\n' }`

And would it be worth having a unit test that the scripts embedded inside `scripts.json` actually do match the platform scripts in the repo?